### DISCUSSION TITLE
Tranformation or processing of CSS files

### DISCUSSION THREAD

**Discussion by viharm** [11ty/eleventy#3044] [Q&A]
Although, this has been partly answeredin #1457, I'm still curious how we could do minor manipulations in CSS files based on some variables from data.

In the linked example, there is a different image for each page, so a `<style>...</style>` block in each html doc is clearly the right solution. However for site-wide styles, that are applied at build time, like fonts & _Bootstrap_ themes, it will be good to have some minimal variables processing in CSS files.

If not, us amateurs will start adding bloat to each page with `<style>...</style>` blocks, thus affecting page load performance. I feel this is important so that the client browsers can cache the CSS files rather than reading the CSS from each page load.

As an added bonus, it will provide a significant enhance of portability of the base path (`pathPrefix`).

Am I missing something? Or is there better way to do this?

Sorry if this has been addressed somewhere else.

Edit: I should have started with this, but "11ty is awesome, and I'm loving it!". Thanks to all the devs and the contributors for such a wonderful resource :-)

**Comment by phineas-pta** [comment]
my workaround is make the css file a template by adding front matter, then override the `permalink` field to output css file

**Reply by pdehaan** [reply]
Yeah, I'll generally either put some/all my CSS in a .njk file and tweak the permalink, or depending on complexity I might just set custom classes on my `<body>` or `<main>` tags if I want to set styles only for specific pages (ie: `<body class="about-page">`, then you can have `.about-page { ... }` in your CSS if you want to set `background-color: salmon;` or whatever).

**Comment by btrem** [comment]
Same here, I have an optional frontmatter item `mainClass` which gets added to the template. Then I can style an individual page (or several). I also have an optional frontmatter item `stylesheet` which contains the filename of a stylesheet that, again, is added to the page via a template if statement in the `head` block. That way I can load an entire stylesheet on a case-by-case basis.
